aptos-move/block-executor/src/view.rs

```diff
@@ -1204,8 +1204,8 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
                     return None;
                 }
 
-                if let Ok(Some(metadata)) = self.get_resource_state_value_metadata(&key) {
-                    return Some(
+                match self.get_resource_state_value_metadata(&key) {
+                    Ok(Some(metadata)) => Some(
                         if let Ok(GroupReadResult::Size(group_size)) =
                             parallel_state.read_group_size(&key, self.txn_idx)
```

Comment:
Could this not get into similar trouble? I think it can.
Moreover, we should maybe document somewhere why the same can't happen to normal (non-group) reads - which should be because the value would be cached? (I suppose, or we have to fix?)